shelley/chain-and-ledger/executable-spec/src/STS/PoolReap.hs

```diff
@@ -50,20 +50,20 @@ poolReapTransition = do
   TRC (pp, PoolreapState us a ds ps, e) <- judgmentContext
 
   let retired = dom $ (ps ^. retiring) ▷ Set.singleton e
-      relevant = retired ◁ (ps ^. pParams)
-      rewardAcnts = Map.intersectionWith (\_ v -> v ^. poolRAcnt) relevant (ps ^. pParams)
       StakePools stPools' = ps ^. stPools
       pr = poolRefunds pp (retired ◁ stPools') (firstSlot e)
-      refunds' = Map.fromList . Map.elems
-               $ Map.intersectionWith (\coin addr -> (addr,coin)) pr rewardAcnts
+      rewardAcnts = Map.map _poolRAcnt $ retired ◁ (ps ^. pParams)
+      rewardAcnts' = Map.fromList . Map.elems $ Map.intersectionWith (,) rewardAcnts pr
 
       domRewards = dom (ds ^. rewards)
-      (refunds, unclaimed') = Map.partitionWithKey (\k _ -> k ∈ domRewards) refunds'
+      (refunds, mRefunds) = Map.partitionWithKey (\k _ -> k ∈ domRewards) rewardAcnts'
+      refunded = sum $ Map.elems refunds
+      unclaimed = sum $ Map.elems mRefunds
 
```

Comment:
looks good, but isn't there a problem in the spec? the spec sums over the range, but the range is a set. Here we do the right thing, i.e., sum over all elements as a list

Reply:
good catch, thanks!